go-controller/pkg/ovn/egressgw.go

```diff
@@ -448,13 +448,10 @@ func (oc *Controller) deleteGWRoutesForPod(name ktypes.NamespacedName, podIPNets
 	for _, podIPNet := range podIPNets {
 		podIP := podIPNet.IP.String()
 		if gwToGr, ok := routeInfo.podExternalRoutes[podIP]; ok {
-			if len(gwToGr) == 0 {
-				delete(routeInfo.podExternalRoutes, podIP)
-				continue
-			}
 			for gw, gr := range gwToGr {
 				oc.deleteOnePodGWRoute(routeInfo, podIP, gw, gr)
 			}
+			delete(routeInfo.podExternalRoutes, podIP)
```

Comment:
this consolidates 4 lines into 1, but I think it might cause some confusion for new folks. They might wonder why we are deleting twice (once in deleteOnePodGWRoute and here). Maybe its fine to just leave it like it was?

Reply:
Dropped this commit.